src/Testcontainers/Containers/Modules/Databases/CosmosDbTestcontainer.cs

```diff
@@ -0,0 +1,50 @@
+namespace DotNet.Testcontainers.Containers
+{
+  using System;
+  using System.Net.Http;
+  using System.Threading;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Configurations;
+  using Microsoft.Extensions.Logging;
+
+  public sealed class CosmosDbTestcontainer : TestcontainerDatabase
+  {
+    internal CosmosDbTestcontainer(ITestcontainersConfiguration configuration, ILogger logger)
+      : base(configuration, logger)
+    {
+    }
+
+    public HttpClient HttpClient
+    {
+      get
+      {
+        var httpMessageHandler = new HttpClientHandler { ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => true, };
+
+        return new HttpClient(new UrlRewriter(this.Hostname, this.Port, httpMessageHandler));
+      }
+    }
+
+    public override string ConnectionString =>
+      $"AccountEndpoint=https://{this.Hostname}:{this.Port};AccountKey={this.Password}";
+
+    private class UrlRewriter : DelegatingHandler
+    {
+      private readonly string host;
+      private readonly int portNumber;
+
+      internal UrlRewriter(string host, int portNumber, HttpMessageHandler innerHandler)
+        : base(innerHandler)
+      {
+        this.host = host;
+        this.portNumber = portNumber;
+      }
+
+      protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+      {
+        request.RequestUri = new Uri($"https://{this.host}:{this.portNumber}{request.RequestUri?.PathAndQuery}");
+        var response = await base.SendAsync(request, cancellationToken);
+        return response;
+      }
+    }
```

Comment:
This looks like a workaround. If we would set the IP of the Docker host as environment variable AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE, then we could get rid of the `UrlRewriter`.

Unfortunately it is not that straightforward to get the IP of the docker host. I tried to set it to `172.17.0.1` as described here: https://stackoverflow.com/questions/48546124/what-is-linux-equivalent-of-host-docker-internal/48547074#48547074. This worked on our Gitlab runner, which uses Socket Bindings like described here: https://docs.gitlab.com/ee/ci/docker/using_docker_build.html, but it failed on my local machine running Docker Desktop 4.11.1 on Windows.

Maybe @HofmeisterAn has an idea how to get the IP of the docker host before the container is started?

Reply:
> This looks like a workaround. If we would set the IP of the Docker host as environment variable AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE, then we could get rid of the `UrlRewriter`.

Yes, I know.  But it wasn't possible when set the override as the ip is unavailable at this point. Since this is only a test support lib I think the workaround is ok.

Reply:
The problem is, when your production code which you want to test still uses the [DocumentClient](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.client.documentclient?view=azure-dotnet), then you are not able to set another `HttpClient`. You can only set the `HttpMessageHandler`.

Of course it would be better to change the production code to use [CosmosClient](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.cosmosclient?view=azure-dotnet), but you want some integration tests using testconainers-dotnet to cover that, before you do that change i guess.

So maybe it would be better to at least provide a property of type `HttpMessageHandler`, instead of `HttpClient`? This would also be compatible with  [HttpClientBuilderExtensions.ConfigurePrimaryHttpMessageHandler](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.httpclientbuilderextensions.configureprimaryhttpmessagehandler?view=dotnet-plat-ext-6.0)

Anyway if we could set AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE to the correct IP this would be the best solution in my opinion.

Reply:
> So maybe it would be better to at least provide a property of type `HttpMessageHandler`, instead of `HttpClient`? This would also be compatible with [HttpClientBuilderExtensions.ConfigurePrimaryHttpMessageHandler](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.httpclientbuilderextensions.configureprimaryhttpmessagehandler?view=dotnet-plat-ext-6.0)
> 
> Anyway if we could set AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE to the correct IP this would be the best solution in my opinion.

This would be the best solution I think. @HofmeisterAn any idea how to get hold of the hosts ip? before configuration.

Reply:
Sorry, I have been on a trip the past days. I think Kafka uses a similar approach. Might be the startup callback useful?

https://github.com/testcontainers/testcontainers-dotnet/blob/997c019428132ebc94dc981b454324e1e7a6f302/src/Testcontainers/Configurations/Modules/MessageBrokers/KafkaTestcontainerConfiguration.cs#L78

Reply:
> Sorry, I have been on a trip the past days. I think Kafka uses a similar approach. Might be the startup callback useful?
> 
 
Yes. This looks promising. I'll try something similar.